### PR TITLE
Allow group admins to manage group images in user portal

### PR DESCRIPTION
**Pull request by aar0nTw**
Story details: https://app.clubhouse.io/infuseai/story/13976

Important functions update:
- Search, now can search all fields in one input.
- Auth, add isGroupAdmin, and customResolver for group Image
- Image query mode => [ALL, SYSTEM_ONLY, GROUP_ONLY]

Screen Shot 2020-12-22 at 4 08 53 PM
Screen Shot 2020-12-22 at 4 09 05 PM
Screen Shot 2020-12-22 at 4 09 23 PM

test image (client/graphql): `ch13976-f097384c`

### CODE REVIEW

**Comment by aar0nTw** [comment]
test image (graphql and client): `ch13976-c1db3d88`

**Comment by popcornylu** [comment]
According to the [spec](https://www.notion.so/infuseai/Spec-Group-Admin-Image-Management-8543a7d4d4cb4378b4e68d1705708ca0#3772c77f19d144409b5bd9eb31b61c46), the group image should not appear in the image list in group in admin portal. 

image

File: packages/graphql-server/src/utils/roles.ts [hunk 1]
```diff
@@ -19,3 +28,12 @@ export const isClient = rule({ cache: 'contextual' })(
     return ctx.role === Role.CLIENT;
   },
 );
+
+export const isGroupAdmin = rule({ cache: 'contextual' })(
+  async (parent, args, ctx, info) => {
+    const currentGroup = await getCurrentGroup(args, ctx);
+    const admins = get(currentGroup, 'attributes.admins', []);
+    const valid = admins.includes(ctx.username);
+    return valid;
```
Comment (ctiml) [inline packages/graphql-server/src/utils/roles.ts:37]: should we just return `admins.includes(ctx.username);`?

File: packages/graphql-server/src/utils/roles.ts [hunk 2]
```diff
@@ -1,5 +1,14 @@
 import { rule } from 'graphql-shield';
+import Boom from 'boom';
```
Comment (ctiml) [inline packages/graphql-server/src/utils/roles.ts:2]: it's unused, should we remove it?

File: packages/graphql-server/src/ee/middlewares/auth.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import { rule, shield, and, or, not } from 'graphql-shield';
-import { isAdmin, isClient, isUser } from '../../utils/roles';
+import { isAdmin, isGroupAdmin, isClient, isUser } from '../../utils/roles';
+import Boom from 'boom';
```
Comment (ctiml) [inline packages/graphql-server/src/ee/middlewares/auth.ts:3]: it's unused?

**Comment by aar0nTw** [comment]
> According to the [spec](https://www.notion.so/infuseai/Spec-Group-Admin-Image-Management-8543a7d4d4cb4378b4e68d1705708ca0#3772c77f19d144409b5bd9eb31b61c46), the group image should not appear in the image list in group in admin portal.
> 
> image

`ch13976-f097384c` do you use the latest image?

**Review by popcornylu — COMMENTED** [review]
Some API questions

**Comment by aar0nTw** [comment]
Latest test image tag(graphql-server / primehub-console): `ch13976-e265eb6d` @popcornylu

**Comment by aar0nTw** [comment]
@popcornylu update tag `ch13976-0f5585df`

File: packages/graphql-server/src/utils/roles.ts [hunk 4]
```diff
@@ -19,3 +27,11 @@ export const isClient = rule({ cache: 'contextual' })(
     return ctx.role === Role.CLIENT;
   },
 );
+
+export const isGroupAdmin = rule({ cache: 'contextual' })(
```
Comment (popcornylu) [inline packages/graphql-server/src/utils/roles.ts:31]: Is `isGroupAdmin` still needed? I think all the ABAC should be all in the resolver.

**Review by popcornylu — COMMENTED** [review]
Please check if we can remove `isGroupAdmin` in auth middleware. Others LGTM.

**Comment by popcornylu** [comment]
LGTM

**Pull request merged**